src/routes/api/cryptor-migrations/migration-router.ts

```diff
@@ -0,0 +1,115 @@
+import { Request, Response } from "express";
+import { WhereOptions } from "sequelize";
+import { Installation } from "models/installation";
+
+const MAX_BATCH_SIZE = 100;
+const MAX_TOTAL_LIMIT = 10_000;
+const SAFE_GUARD_LOOP_COUNT = 100;
+
+export const CryptorMigrationInstallationPost = async (req: Request, res: Response): Promise<void> => {
+
+	const startTime = Date.now();
+
+	let batchSize = parseInt(req.query.batchSize as string);
+	let totalLimit = parseInt(req.query.totalLimit as string);
+	const jiraHost = req.query.jiraHost as string;
+
+	req.log = req.log.child({
+		operation: "migrate-installations-shared-secret",
+		batchSize,
+		totalLimit,
+		jiraHost
+	});
+
+	req.log.info(`Received to migration installation sharedSecret for`);
+	if (jiraHost) {
+		if (batchSize || totalLimit) {
+			req.log.warn("Invalid params");
+			res.status(400).send("Since 'jiraHost' is provided, we are only migrating one record for a time. Please do not specify 'batchSize' nor 'totalLimit'");
+			return;
+		}
+		//updating single entry, hard code it to one
+		totalLimit = 1;
+		batchSize = 1;
+		req.log.fields.batchSize = 1;
+		req.log.fields.totalLimit = 1;
+		req.log.debug(`Found jiraHost, now hardcoded totalLimit and batchSize to 1`);
+	} else {
+		if (isNaN(batchSize) || batchSize < 0 || batchSize > MAX_BATCH_SIZE) {
+			req.log.warn("Invalid params");
+			res.status(400).send(`Please provide a valid batchSize between 0 and ${MAX_BATCH_SIZE}, got ${batchSize}`);
+			return;
+		}
+		if (isNaN(totalLimit) || totalLimit < 0 || totalLimit > MAX_BATCH_SIZE) {
+			req.log.warn("Invalid params");
+			res.status(400).send(`Please provide a valid totalLimit between 0 and ${MAX_TOTAL_LIMIT}, got ${totalLimit}`);
+			return;
+		}
+	}
+
+	const toMigrateWhere: WhereOptions = {};
+	if (jiraHost) {
+		toMigrateWhere.jiraHost = jiraHost;
+	} else {
+		toMigrateWhere.encryptedSharedSecret = null;
+	}
+
+	req.log.info("About to start migrating installation sharedSecret");
+	let count = 0;
+	let safeGuardLoopCount = 0;
+	while (count < totalLimit) {
```

Comment:
you should _never_ use a while loop with a DB operation

Reply:
What do you suggest for migrating xxx amount of records? Say we want to migrate 1000 records everytime we call this api, you mean we shall pull that 1000 records from db in one go?

Reply:
fixed. will look into testing and see whether we need BB pipeline looping.